pkg/util/storage/manager.go

```diff
@@ -33,6 +35,32 @@ func NewManager(env env.Core, subscriptionID string, authorizer autorest.Authori
 	}
 }
 
+func getCorrectErrWhenTooManyRequests(err error) error {
+	detailedError, ok := err.(autorest.DetailedError)
+	if !ok {
+		return err
+	}
+
+	if detailedError.StatusCode != http.StatusTooManyRequests {
+		return err
+	} else {
```

Comment:
One last comment before LGTM, the else is not necessary due to the previous return (we can avoid one nested level then). One of the benefits of having a separate function for this 🙂